dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/GenericStore.java

```diff
@@ -72,9 +72,11 @@
      *
      * @param id the object identifier.
      * @return the object identified by the given identifier.
+     * @throws IllegalQueryException when no such object exists in the database
      */
     @Nonnull
-    T get( long id );
+    T get( long id )
+        throws IllegalQueryException;
```

Comment:
I thought this always would have thrown some low level exception but it actually would not. I changed this now so it either returns a value or throws an `IllegalQueryException`.

Reply:
Should we add the `find` methods as a part of this refactoring ? 
I think in some places we might expect `null` result instead of the exception.

Reply:
reverted this.